packages/aws-cdk-lib/aws-stepfunctions/test/custom-state.test.ts

```diff
@@ -309,4 +310,145 @@ describe('Custom State', () => {
       },
     );
   });
+
+  test('expect retry to merge when specifying strategy inline and through construct', () => {
```

Comment:
(non-blocking) this is already covered as part of this test:

 https://github.com/aws/aws-cdk/blob/8ede40976392893a4a7c0e90f3d6638209670fb6/packages/aws-cdk-lib/aws-stepfunctions/test/custom-state.test.ts#L168